go/arrow/schema.go

```diff
@@ -141,28 +137,25 @@ func (sc *Schema) Metadata() Metadata { return sc.meta }
 func (sc *Schema) Fields() []Field    { return sc.fields }
 func (sc *Schema) Field(i int) Field  { return sc.fields[i] }
 
-func (sc *Schema) FieldByName(n string) (Field, bool) {
-	i, ok := sc.index[n]
+func (sc *Schema) FieldsByName(n string) ([]Field, bool) {
+	indices, ok := sc.index[n]
 	if !ok {
-		return Field{}, ok
+		return nil, ok
 	}
-	return sc.fields[i], ok
-}
-
-// FieldIndex returns the index of the named field or -1.
-func (sc *Schema) FieldIndex(n string) int {
-	i, ok := sc.index[n]
-	if !ok {
-		return -1
+	fields := make([]Field, 0, len(indices))
+	for _, v := range indices {
+		fields = append(fields, sc.fields[v])
 	}
-	return i
+	return fields, ok
 }
 
-func (sc *Schema) HasField(n string) bool {
-	return sc.FieldIndex(n) >= 0
+// FieldsIndex returns the indices of the named field or nil.
+func (sc *Schema) FieldsIndex(n string) []int {
```

Comment:
```suggestion
func (sc *Schema) FieldIndices(n string) []int {
```
?

Reply:
that's indeed the naming convention C++ uses.
let's do that.